From 45d480ea5232a4a6d9edcd1664e89c82e4451b6f Mon Sep 17 00:00:00 2001
From: Francois Daoust <fd@tidoust.net>
Date: Mon, 17 Aug 2026 18:31:50 +0200
Subject: [PATCH] Drop interfaces redefined in css-fonts-5

The `CSSFontFaceDescriptors` and `CSSFontFaceRule` interfaces get redefined in
CSS Fonts Level 5, creating duplicates. The patch will likely be needed until
CSS Fonts Level 5 becomes a full spec.
---
 ed/idl/css-fonts.idl | 38 --------------------------------------
 1 file changed, 38 deletions(-)

diff --git a/ed/idl/css-fonts.idl b/ed/idl/css-fonts.idl
index 0a2136fdbd..b925d4d6a6 100644
--- a/ed/idl/css-fonts.idl
+++ b/ed/idl/css-fonts.idl
@@ -3,44 +3,6 @@
 // (https://github.com/w3c/webref)
 // Source: CSS Fonts Module Level 4 (https://drafts.csswg.org/css-fonts-4/)
 
-[Exposed=Window]
-interface CSSFontFaceDescriptors : CSSStyleDeclaration {
-  attribute [LegacyNullToEmptyString] CSSOMString src;
-  attribute [LegacyNullToEmptyString] CSSOMString fontFamily;
-  attribute [LegacyNullToEmptyString] CSSOMString font-family;
-  attribute [LegacyNullToEmptyString] CSSOMString fontStyle;
-  attribute [LegacyNullToEmptyString] CSSOMString font-style;
-  attribute [LegacyNullToEmptyString] CSSOMString fontWeight;
-  attribute [LegacyNullToEmptyString] CSSOMString font-weight;
-  attribute [LegacyNullToEmptyString] CSSOMString fontStretch;
-  attribute [LegacyNullToEmptyString] CSSOMString font-stretch;
-  attribute [LegacyNullToEmptyString] CSSOMString fontWidth;
-  attribute [LegacyNullToEmptyString] CSSOMString font-width;
-  attribute [LegacyNullToEmptyString] CSSOMString unicodeRange;
-  attribute [LegacyNullToEmptyString] CSSOMString unicode-range;
-  attribute [LegacyNullToEmptyString] CSSOMString fontFeatureSettings;
-  attribute [LegacyNullToEmptyString] CSSOMString font-feature-settings;
-  attribute [LegacyNullToEmptyString] CSSOMString fontVariationSettings;
-  attribute [LegacyNullToEmptyString] CSSOMString font-variation-settings;
-  attribute [LegacyNullToEmptyString] CSSOMString fontNamedInstance;
-  attribute [LegacyNullToEmptyString] CSSOMString font-named-instance;
-  attribute [LegacyNullToEmptyString] CSSOMString fontDisplay;
-  attribute [LegacyNullToEmptyString] CSSOMString font-display;
-  attribute [LegacyNullToEmptyString] CSSOMString fontLanguageOverride;
-  attribute [LegacyNullToEmptyString] CSSOMString font-language-override;
-  attribute [LegacyNullToEmptyString] CSSOMString ascentOverride;
-  attribute [LegacyNullToEmptyString] CSSOMString ascent-override;
-  attribute [LegacyNullToEmptyString] CSSOMString descentOverride;
-  attribute [LegacyNullToEmptyString] CSSOMString descent-override;
-  attribute [LegacyNullToEmptyString] CSSOMString lineGapOverride;
-  attribute [LegacyNullToEmptyString] CSSOMString line-gap-override;
-};
-
-[Exposed=Window]
-interface CSSFontFaceRule : CSSRule {
-  [SameObject, PutForwards=cssText] readonly attribute CSSFontFaceDescriptors style;
-};
-
 partial interface CSSRule {
   const unsigned short FONT_FEATURE_VALUES_RULE = 14;
 };
-- 
2.55.0

